mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -0,0 +1,202 @@
+
+            inflight_uploads = {}
+            for staged_upload, write_credential_info in zip(
+                staged_upload_chunk, write_credential_infos
+            ):
+                upload_future = self.thread_pool.submit(
+                    self._upload_to_cloud,
+                    cloud_credential_info=write_credential_info,
+                    src_file_path=staged_upload.src_file_path,
+                    artifact_path=staged_upload.artifact_path,
+                )
+                inflight_uploads[staged_upload.src_file_path] = upload_future
+
+            for src_file_path, upload_future in inflight_uploads.items():
+                try:
+                    upload_future.result()
+                except Exception as e:
+                    failed_uploads[src_file_path] = repr(e)
+
+        # Iterate over batches of files and upload them
+        for chunk in chunk_list(staged_uploads, _ARTIFACT_UPLOAD_BATCH_SIZE):
+            get_creds_and_upload(chunk)
+
+        if len(failed_uploads) > 0:
+            raise MlflowException(
+                message=(
+                    "The following failures occurred while uploading one or more artifacts"
+                    f" to {self.artifact_uri}: {failed_uploads}"
+                )
+            )
+
+    @abstractmethod
+    def _get_write_credential_infos(self, paths):
+        """
+        For a batch of local files, get the write credentials for each file, which include
+        a presigned URL per file
+
+        Return a list of CredentialInfo objects, one for each file.
+        """
+        pass
+
+    @abstractmethod
+    def _upload_to_cloud(self, cloud_credential_info, src_file_path, artifact_path):
+        """
+        Upload a single file to the cloud.
+        :param cloud_credential_info: ArtifactCredentialInfo containing presigned URL for the current file
+                                      Note: in S3 this gets ignored
+        :param src_file_path:
+        :param artifact_path:
+        :return:
+        """
+        pass
+
+    # Read APIs
+
+    def _extract_headers_from_credentials(self, headers):
+        """
+        :return: A python dictionary of http headers converted from the protobuf credentials
+        """
+        return {header.name: header.value for header in headers}
+
+    def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_path):
+        read_credentials = self._get_read_credential_infos(remote_file_path)
+        # Read credentials for only one file were requested. So we expected only one value in
+        # the response.
+        assert len(read_credentials) == 1
+        cloud_credential_info = read_credentials[0]
+        # from mlflow.utils.databricks_utils import get_databricks_env_vars
+
+        try:
+            parallel_download_subproc_env = os.environ.copy()
+            # parallel_download_subproc_env.update(
+            #     get_databricks_env_vars(self.databricks_profile_uri)
+            # )
```

Comment:
QQ: @harupy what was the purpose of copying environment variables to sub-processes here? Is it still necessary?